bindings/pyroot_experimental/PyROOT/src/PyzCppHelpers.cxx

```diff
@@ -0,0 +1,29 @@
+// Author: Danilo Piparo CERN  08/2018
+
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#include "PyzCppHelpers.hxx"
+
+#include "CPyCppyy.h"
+#include "CPPInstance.h"
+#include "TClass.h"
+
+PyObject *CallPyObjMethod(PyObject *obj, const char *meth, PyObject *arg1)
+{
+   // Helper; call method with signature: obj->meth(arg1).
+   Py_INCREF(obj);
+   PyObject *result = PyObject_CallMethod(obj, const_cast<char *>(meth), const_cast<char *>("O"), arg1);
+   Py_DECREF(obj);
+   return result;
+}
+
+TClass *OP2TCLASS(CPyCppyy::CPPInstance *pyobj)
```

Comment:
This method is actually defined in `TTreePyz.cxx` too [here](https://github.com/root-project/root/blob/360b252c6cbab44b5e742bddc236500992d8d15d/bindings/pyroot_experimental/PyROOT/src/TTreePyz.cxx#L24) with a different name to make it comply with the coding guidelines. It is a good thing you factored it out already, we just need now to change its name here and invoke it from `TTreePyz.cxx` too.

Reply:
Ok. I will take care of that, maybe in a separate PR which comes immediately after this one?

Reply:
Perhaps if we merge this PR first, what I can do is rebase the PR about `SetBranchAddress` and modify it to invoke the helper method instead.

Reply:
Actually, #2476 was merged before, so after we rebase and merge this PR I will add the corresponding changes to make `TTreePyz.cxx` rely on the helper method.

Reply:
Good for me.